frappe/database/database.py

```diff
@@ -85,7 +85,7 @@ def get_database_size(self):
 
 	def sql(self, query, values=(), as_dict = 0, as_list = 0, formatted = 0,
 			debug=0, ignore_ddl=0, as_utf8=0, auto_commit=0, update=None,
-			explain=False, run=True, pluck=False):
+			explain=False, run=True, pluck=False, **kwargs):
```

Comment:
I'm not fully onboard accepting and passing all kwargs to `frappe.has_permission` but okay I don't think this would cause it to behave in unintended ways...Would it?

@sagarvora What do you think?

Reply:
IMHO, it's just bad logic to pass the arguments for one function inside another. Can this be refactored?
 `frappe.db.sql` is supposed to be raw (without permission checks). I understand that we want to check permissions, but I don't think db.py is the place for that. Can someone explain the use-case please?

Reply:
So the reason to implement permissions here is to resolve the whole multiple database APIs issues with frappe, slowly moving to a single database API as query building in db_query is quite complex and doesn't support a lot of SQL options.

Reply:
I get that, but if you add permissions to `frappe.db` it will eventually become `db_query`
Do we intend to completely scrap `db_query` some day?

Reply:
Perhaps you should be making `db_query` unified instead?

Reply:
Eventually, `db.sql` will become something that works only if your code is written in the acceptable DB and will be heavily discouraged. And that's okay. But changing what the function means.. I'm not sure how well it'd go.

Reply:
Makes sense 🤔 thinking about moving this to frappe.qb objects only for now.